actors/evm/tests/calc.rs

```diff
@@ -0,0 +1,159 @@
+mod asm;
+
+use evm::interpreter::U256;
+use fil_actor_evm as evm;
+use fil_actors_runtime::test_utils::*;
+use fvm_ipld_encoding::RawBytes;
+
+#[allow(dead_code)]
+pub fn magic_calc_contract() -> Vec<u8> {
+    let init = r#"
+push1 0x42  # magic value
+push1 0x00  # key of magic value
+sstore
+"#;
+    let body = r#"
+# method dispatch:
+# - 0x00000000 -> magic value
+# - 0x00000001 -> ADD arg, magic value
+# - 0x00000002 -> MUL arg, magic value
+
+push1 0x00
+calldataload
+push1 0xe0   # 28 byte shift == 224 bits
+shr
+
+# 0x00 -> jmp get_magic
+dup1
+iszero
+%push(get_magic)
+jumpi
+
+# 0x01 -> jmp add_magic
+dup1
+push1 0x01
+eq
+%push(add_magic)
+jumpi
+
+# 0x02 -> jmp mul_magic
+dup1
+push1 0x02
+eq
+%push(mul_magic)
+jumpi
```

Comment:
I think dispatch is going to get repetitive, so over time we can develop a macro for it.